Assets/HealthReplenish.cs

```diff
@@ -0,0 +1,21 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HealthReplenish : MonoBehaviour
+{
+
+    void OnTriggerEnter(Collider other)
+    {
+        // Get the PlayerHealth component from the player
+        PlayerHealth playerHealth = other.GetComponent<PlayerHealth>();
```

Comment:
You should check the player tag first so you don't have to grab a component for every collision:

`if(other.compareTag("Player")`

Move this script into the Script folder